### PR TITLE
#1168 Refactor KeyService

### LINKED ISSUE #1168: KeyService refactor
- Instead of passing `currentUserEmail` in constructor, it should be passed directly into the methods that need it as argument (non-nilable)
- `coreService` should be called `core`
- `getPrvKeyDetails` should be removed. Instead, the key parsing code should be moved to KeyMethods to become `KeyMethods.parseKeys(armored: [String]) -> [KeyDetails]`.  Then whoever currently calls `getPrvKeyDetails` should call `storage.getKeypairs` then `KeyMethods.parseKeys`.
- the pass phrase related code in `getPrvKeyInfo` and `getSigningKey` both does the same thing - adds pass phrases, if available, to the key. It can be abstracted into a private method.

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR refactored KeyService

close #1168 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by tomholub — COMMENTED** [review]
looks good - see comment

File: FlowCryptAppTests/Functionality/Services/Key Services/KeyServiceTests.swift [hunk 1]
```diff
@@ -53,12 +53,11 @@ final class KeyServiceTests: XCTestCase {
...
         let keyService = KeyService(
             storage: keyStorage,
-            passPhraseService: passPhraseService,
-            currentUserEmail: { "bill@test.com" }
+            passPhraseService: passPhraseService
         )
 
         // act
-        let result = try await keyService.getSigningKey()
+        let result = try await keyService.getSigningKey(email: "bill@test.com")
```
[the remark this answers is not in the document]
Comment (ioanmo226) [reply FlowCryptAppTests/Functionality/Services/Key Services/KeyServiceTests.swift]: Ok

[the remark this answers is not in the document]
Comment (ioanmo226) [reply FlowCryptAppTests/Functionality/Services/Key Services/KeyServiceTests.swift]: @tomholub When I changed `bill@test.com` to `bill@example.test`, `result` got nil when I call `getSigningKey` function.
I think we need to add signing key info for `bill@example.test` to key serve?

**Review by tomholub — APPROVED** [review]
Code looks good to me, but tests need to be investigated - maybe something broke.

File: FlowCryptAppTests/Functionality/Services/Key Services/KeyServiceTests.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (mobilestar1) [reply FlowCryptAppTests/Functionality/Services/Key Services/KeyServiceTests.swift]: @tomholub plaese check above message

[the remark this answers is not in the document]
Comment (ioanmo226) [reply FlowCryptAppTests/Functionality/Services/Key Services/KeyServiceTests.swift]: @tomholub Please check above message.

[the remark this answers is not in the document]
Comment (tomholub) [reply FlowCryptAppTests/Functionality/Services/Key Services/KeyServiceTests.swift]: It's not a production keyserver that it's using, anyway. You'd have to check where was it originally coming from. I thought all of the key sources were mocked for this test, configured in the test itself.

**Review by tomholub — APPROVED** [review]

**Pull request merged**